### ISSUE TITLE
Any plan to add SoftClipping and SmoothMax

### ISSUE THREAD

**Issue by dmilos** [tensorflow/tensorflow#26259]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using):
- Are you willing to contribute it (Yes/No):
Yes.
**Describe the feature and the current behavior/state.**
SoftClipping for activation. [https://en.wikipedia.org/wiki/Activation_function]
SmootMax for pooling. [https://en.wikipedia.org/wiki/Smooth_maximum]
**Will this change the current api? How?**
No. This is new feature.
**Who will benefit with this feature?**
Everybody. 
SmoothClipping is smoot function and can not suffer from vanishing/exploding gradients. 
So with this we will have faster learning. https://github.com/tiny-dnn/tiny-dnn/pull/1014.  Only 10 layers achieve 99.35% on MNIST set, Comparing to original 12 and 99.0%
**Any Other info.**

**Comment by rmothukuru** [comment]
@dmilos,
Sorry for the delayed response. [Relu Activation Function](https://arxiv.org/pdf/1505.00853.pdf%C3%A3%E2%82%AC%E2%80%9AReLU) seems to be good for handling the problem of **`Vanishing and Exploding Gradients`**.

Can you please let us know how efficient are the **`activation functions`** that you proposed compared to the existing **`non-saturated activation functions`** like [Relu](https://www.tensorflow.org/api_docs/python/tf/keras/activations/relu), [Elu](https://www.tensorflow.org/api_docs/python/tf/keras/activations/elu), [Selu](https://www.tensorflow.org/api_docs/python/tf/keras/activations/selu)? Thanks!

**Comment by dmilos** [comment]
SmoothClipping is smooth function and can not suffer from vanishing/exploding gradients.
So with this we will have faster learning. tiny-dnn/tiny-dnn#1014. Only 10 layers achieve 99.35% on MNIST set, Comparing to original 12 and 99.0%

**Comment by dmilos** [comment]
For more details ind dept for  Soft Clipping see:
Neural Network-Based Approach to Phase Space Integration
M. D. Klimek1,2*, M. Perelstein1
1 Laboratory for Elementary Particle Physics, Cornell University, Ithaca, NY, USA
2 Department of Physics, Korea University, Seoul, Republic of Korea
* klimek@cornell.edu
August 18, 2020

**Comment by dmilos** [comment]
Appears that answer is: no.
Thanks.

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
